src/click/core.py

```diff
@@ -2186,7 +2155,12 @@ def to_info_dict(self) -> dict[str, t.Any]:
             "required": self.required,
             "nargs": self.nargs,
             "multiple": self.multiple,
-            "default": self.default,
+            # XXX Normalize the default to None if UNSET. This is acceptable for
+            # now as to_info_dict() has been designed to be used for documentation
+            # purposes, and UNSET is not a user-facing value.
+            # TODO: Make UNSET a first class citizen in Click 9.0, remove this
```

Comment:
I don't think we should make `UNSET` public. It should remain an implementation detail to make the internal tracking work, but users should never be passing or receiving it.